gateway/src/inference/providers/anthropic.rs

```diff
@@ -518,6 +539,62 @@ fn prepare_messages(messages: Vec<AnthropicMessage>) -> Result<Vec<AnthropicMess
     Ok(consolidated_messages)
 }
 
+fn prefill_json_message(messages: Vec<AnthropicMessage>) -> Vec<AnthropicMessage> {
+    let mut messages = messages;
+    // Add a JSON-prefill message for Anthropic's JSON mode
+    messages.push(AnthropicMessage {
+        role: AnthropicRole::Assistant,
+        content: vec![AnthropicMessageContent::Text {
+            text: "Here is the JSON requested:\n{",
+        }],
+    });
+    messages
+}
+
+pub(crate) fn prefill_json_response(
+    content: Vec<ContentBlock>,
+) -> Result<Vec<ContentBlock>, Error> {
+    // Check if the content is a single text block
+    if content.len() == 1 {
+        if let ContentBlock::Text(text) = &content[0] {
+            // If it's a single text block, add a "{" to the beginning
+            return Ok(vec![ContentBlock::Text(Text {
+                text: format!("{{{}", text.text.trim()),
+            })]);
+        }
+    }
+    // If it's not a single text block, return content as-is but log an error
+    Error::OutputParsing {
+        message: "Expected a single text block in the response from Anthropic".to_string(),
+        raw_output: serde_json::to_string(&content).map_err(|e| Error::Inference {
+            message: format!("Error serializing content as JSON: {e}. This should never happen. Please file a bug report: https://github.com/tensorzero/tensorzero/issues/new"),
+        })?,
+    }
+    .log();
+    Ok(content)
+}
+
+pub(crate) fn prefill_json_chunk_response(
+    chunk: ProviderInferenceResponseChunk,
+) -> ProviderInferenceResponseChunk {
+    let mut chunk = chunk;
+    if chunk.content.is_empty() {
+        chunk.content = vec![ContentBlockChunk::Text(TextChunk {
+            text: "{".to_string(),
+            id: chunk.inference_id.to_string(),
+        })];
+    } else if chunk.content.len() == 1 {
+        if let ContentBlockChunk::Text(TextChunk { text, .. }) = &chunk.content[0] {
+            // Add a "{" to the beginning of the text
+            chunk.content = vec![ContentBlockChunk::Text(TextChunk {
+                text: format!("{{{}", text.trim()),
```

Comment:
You can't trim chunks. Imagine chunks `{"msg": "Hello "` + `world ".}`

Left-trim only. (Above should be ok but might as well do it right.)

Reply:
Add this test case